Core/Core/Discussions/DiscussionReplyViewController.swift

```diff
@@ -0,0 +1,286 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2020-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import UIKit
+
+public class DiscussionReplyViewController: UIViewController, CoreWebViewLinkDelegate, ErrorViewController, RichContentEditorDelegate {
+    lazy var contentHeight = webView.heightAnchor.constraint(equalToConstant: 0)
+    var contentHeightObs: NSKeyValueObservation?
+    @IBOutlet weak var editorContainer: UIView!
+    @IBOutlet weak var loadingView: CircleProgressView!
+    @IBOutlet weak var keyboardSpace: NSLayoutConstraint!
+    @IBOutlet weak var scrollView: UIScrollView!
+    let titleSubtitleView = TitleSubtitleView.create()
+    @IBOutlet weak var viewMoreButton: UIButton!
+    var webView = CoreWebView()
+    @IBOutlet weak var webViewContainer: UIView!
+    @IBOutlet var webViewHeight: NSLayoutConstraint!
+
+    lazy var attachButton: UIBarButtonItem = {
```

Comment:
The existing addBadge on UIButton extension did not give us something that looks anything like this. extra stuff in UIButton is also something a storyboard/xib cannot do.